src/test/java/uk/nhs/hee/trainee/details/service/TrainingNumberGeneratorTest.java

```diff
@@ -393,6 +393,66 @@ void shouldThrowExceptionPopulatingTrainingNumberWhenParentOrganizationNull(Stri
         is("Unable to calculate the parent organization."));
   }
 
+  @Test
+  void shouldUseTsdPrefixWhenMilitaryProgrammeMembershipType() {
+    TraineeProfileDto profile = new TraineeProfileDto();
+    PersonalDetailsDto personalDetails = new PersonalDetailsDto();
+    personalDetails.setGmcNumber(GMC_NUMBER);
+    profile.setPersonalDetails(personalDetails);
+
+    ProgrammeMembershipDto pm = new ProgrammeMembershipDto();
+    pm.setProgrammeMembershipType("MILITARY");
+    pm.setManagingDeanery(OWNER_NAME);
+    pm.setProgrammeName(PROGRAMME_NAME);
+    pm.setProgrammeNumber(PROGRAMME_NUMBER);
+    pm.setTrainingPathway(TRAINING_PATHWAY);
+    pm.setStartDate(NOW);
+    profile.setProgrammeMemberships(List.of(pm));
+
+    CurriculumDto curriculum1 = new CurriculumDto();
+    curriculum1.setCurriculumSpecialtyCode("ABC");
+    curriculum1.setCurriculumSubType(CURRICULUM_SUB_TYPE_MC);
+    curriculum1.setCurriculumStartDate(PAST);
+    curriculum1.setCurriculumEndDate(FUTURE);
+
+    pm.setCurricula(List.of(curriculum1));
+
+    service.populateTrainingNumbers(profile);
+
+    assertThat("Unexpected training number.", pm.getTrainingNumber(),
+        is("TSD/ABC/1234567/D"));
+  }
+
+  @Test
+  void shouldNotForceTsdPrefixWhenNotMilitaryProgrammeMembershipType() {
```

Comment:
I'd be tempted to get rid of this one, since it doesn't add anything above and beyond existing tests that verify the prefix for each LO.

Reply:
Fine by me: I think I was pre-emptively trying to handle Sonar's rather stricter test coverage rules 😁